src/xdocs/config_coding.xml

```diff
@@ -36,17 +36,56 @@ int[] a = new int[]
         </source>
 
         <p>
-          The check allows leaving out the comma at the end if both the left and right curly brackets
-          are on the same line.
+          The check allows leaving out the comma at the end if either left
+          or right curly braces are on the same line as the last element of the array.
         </p>
         <source>
 return new int[] { 0 };
+return new int[] { 0
+        };
+return new int[] {
+        0 };
         </source>
 
         <p>
           Rationale: Putting this comma in makes it easier to change the order
-          of the elements or add new elements on the end.
+          of the elements or add new elements on the end. Main benefit of a trailing
+          comma is that when you add new entry to an array, no surrounding lines are changed.
         </p>
+        <source>
+{
+    100000000000000000000,
+    200000000000000000000, // OK
+}
+
+{
+    100000000000000000000,
+    200000000000000000000,
+    300000000000000000000,  // Just this line added, no other changes
+}
+        </source>
+        <p>
+          If closing brace is on the same line as training comma, this benefit is gone:
```

Comment:
lets make a note that this Check does not demand certain location of curly braces, so following two cases does not produce an violation.